### PR TITLE
Robot name exercise

### PR DESCRIPTION
**Pull request by plasticwrapper**
here we go...

### CODE REVIEW

File: robot_name/robot_name_test.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def test_robot_name_conflict
+    assert_raises NameCollisionError do
+      generator = -> { 'AA111' }
+      Robot.new(name_generator: generator)
+      Robot.new(name_generator: generator)
+    end
+  end
+end
```
Comment (jmmastey) [inline robot_name/robot_name_test.rb]: If you're going to add tests, use rspec :)

File: robot_name/robot_name.rb [hunk 2]
```diff
@@ -3,23 +3,52 @@ class NameCollisionError < RuntimeError; end
...
     @@registry ||= []
     @name_generator = args[:name_generator]
+    @name = pick_a_name
+
+    raise NameCollisionError, NAME_ERROR_MSG if name_issues?(@name)
+
+    @@registry << @name
+  end
 
+  def pick_a_name
```
Comment (jmmastey) [inline robot_name/robot_name.rb]: Most if not all of the methods under this should be private.

File: robot_name/robot_name.rb [hunk 3]
```diff
@@ -3,23 +3,52 @@ class NameCollisionError < RuntimeError; end
...
     if @name_generator
-      @name = @name_generator.call
+      @name_generator.call
     else
-      generate_char = -> { ('A'..'Z').to_a.sample }
-      generate_num = -> { rand(10) }
-
-      @name = "#{generate_char.call}#{generate_char.call}#{generate_num.call}#{generate_num.call}#{generate_num.call}"
+      generate_name
     end
```
Comment (jmmastey) [inline robot_name/robot_name.rb]: You can use the `Kernel#method` method to turn `generate_name` into a proc, which would eliminate a conditional here.

**Comment by jmmastey** [comment]
The stabby lambdas are not awesome where they are, but otherwise all is well.

File: robot_name/robot_name.rb (hunk 2 above)
Comment (plasticwrapper) [inline robot_name/robot_name.rb]: agreed.

File: robot_name/robot_name_test.rb (hunk 1 above)
Comment (plasticwrapper) [inline robot_name/robot_name_test.rb]: whateva whateva!  I do what I want!

Fine.

**Comment by plasticwrapper** [comment]
agreed on the stabby lambdas... I wanted to just get rid of them, but decided to leave them in and move on.

**Pull request closed without merging**